extensions/wikia/UserRenameTool/tasks/SingleWikiRename.class.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace UserRenameTool\Tasks;
+
+class SingleWikiRename extends WikiRenameBase {
+	const SCRIPT_TEMPLATE = "SERVER_ID=%s php %s/extensions/wikia/maintenance/RenameUser_local.php %s";
+
+	private $cityId;
+	private $command;
+	private $exitCode = null;
+
+	/**
+	 * Rename a user on a single wiki
+	 *
+	 * @param int $cityId
+	 * @param array $params See comment in WikiRenameBase::params for more details.
+	 * @return bool
+	 */
+	public function run( $cityId, array $params ) {
+		$this->cityId = $cityId;
+		$this->params = $params;
+
+		$this->setupProcessLocal();
+		$this->buildCommand();
+
+		try {
+			$this->runCommand();
+			$this->logFinish();
+			$this->cleanup();
+		} catch ( \Exception $e ) {
+			$this->error( "error while renaming user", [
+				'message' => $e->getMessage(),
+				'stack' => $e->getTraceAsString(),
+			] );
+			return false;
+		}
+
+		$this->recordWikiRenamed( $this->cityId );
+
+		return true;
+	}
+
+	private function buildCommand() {
+		global $IP;
+
+		$opts = $this->getShellOptions( $this->params );
+
+		// Run the maintenance script for each wiki
+		$this->command = sprintf( self::SCRIPT_TEMPLATE, $this->cityId, $IP, $opts );
+	}
+
+	private function getShellOptions( $params ) {
+		$renameIP = !empty( $params['rename_ip'] );
+
+		$opts = [
+			'rename-user-id' => $params['rename_user_id'],
+			'requestor-id' => $params['requestor_id'],
+			'reason' => $params['reason'],
+			'start-time' => $params['start_time'],
+		];
+
+		if ( $renameIP ) {
+			$opts['rename-old-name'] = $params['rename_old_name'];
+			$opts['rename-new-name'] = $params['rename_new_name'];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "rename_new_name" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)